test/helpers/json_helper.rb

```diff
@@ -0,0 +1,33 @@
+require 'test_helper'
+
+module TestHelpers
+  #
+  # JSON Helpers
+  #
+  module JsonHelper
+    #
+    # POSTs a hash data as JSON with content-type "application/json"
+    #
+    def post_json(endpoint, data)
+      post endpoint, data.to_json, 'CONTENT_TYPE' => 'application/json'
+    end
+
+    #
+    # PUTs a hash data as JSON with content-type "application/json"
+    #
+    def put_json(endpoint, data)
+      put endpoint, data.to_json, 'CONTENT_TYPE' => 'application/json'
+    end
+
+    #
+    # Assert that a JSON response matches the model and keys provided
+    #
+    def assert_json_matches_model(response_json, model, keys)
+      keys.each { |k| assert response_json.has_key?(k), "Response has key #{k}"}
+      keys.each { |k| assert_equal model[k], response_json[k], "Values for key #{k} match" }
+    end
+
+    module_function :assert_json_matches_model
+    module_function :post_json
```

Comment:
Whoops! Yes that's right 

Sent from my iPhone

> On 8 Jul 2016, at 9:09 PM, Andrew Cain <notifications@github.com> wrote:
> 
> In test/helpers/json_helper.rb:
> 
> > +    # PUTs a hash data as JSON with content-type "application/json"
> > +    #
> > +    def put_json(endpoint, data)
> > +      put endpoint, data.to_json, 'CONTENT_TYPE' => 'application/json'
> > +    end
> > +
> > +    #
> > +    # Assert that a JSON response matches the model and keys provided
> > +    #
> > +    def assert_json_matches_model(response_json, model, keys)
> > +      keys.each { |k| assert response_json.has_key?(k), "Response has key #{k}"}
> > +      keys.each { |k| assert_equal model[k], response_json[k], "Values for key #{k} match" }
> > +    end
> > +
> > +    module_function :assert_json_matches_model
> > +    module_function :post_json
> @alexcu Should put_json also be exported?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>